nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/model/CandidateAggregate.java

```diff
@@ -0,0 +1,41 @@
+package no.sikt.nva.nvi.common.db.model;
+
+import java.util.Collection;
+import java.util.Optional;
+import no.sikt.nva.nvi.common.db.ApprovalStatusDao;
+import no.sikt.nva.nvi.common.db.CandidateDao;
+import no.sikt.nva.nvi.common.db.Dao;
+import no.sikt.nva.nvi.common.db.NoteDao;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Wrapper class for candidate data that should be read in a single transaction. */
+public record CandidateAggregate(
+    CandidateDao candidate, Collection<ApprovalStatusDao> approvals, Collection<NoteDao> notes) {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CandidateAggregate.class);
+  private static final int ONE = 1;
+
+  public static Optional<CandidateAggregate> fromQueryResponse(Collection<Dao> items) {
+    var candidates =
+        items.stream()
+            .filter(CandidateDao.class::isInstance)
+            .map(CandidateDao.class::cast)
+            .toList();
+    if (candidates.size() > ONE) {
+      LOGGER.error("Expected single candidate, but found {}", candidates.size());
+      throw new IllegalStateException("Multiple candidates returned by aggregate query");
+    }
```

Comment:
Is this exception tested? It should not be possible to insert item to dynamo with the same partition key and sort key.
If we mean we can have this condition, than maybe test it. 
<img width="913" height="45" alt="image" src="https://github.com/user-attachments/assets/33bd5395-2dce-4970-8a63-9135fa826c84" />